### PR TITLE
Check that parent is defined before getting its metrics.  mathjax/MathJax#2191

### PR DESCRIPTION
**Pull request by dpvc**
If a typeset expression has been removed from the page, typesetting again can cause an error when the metrics are being determined.  This PR checks that the location where the metrics are being taken has a parent element before trying to use it.

Resolves issue mathjax/MathJax#2191.

### CODE REVIEW

**Comment by dpvc** [comment]
I added code to check that math item metrics are not being remeasured when they have already been measured, since it is not efficient to remeasure old math when new math is being processed.

**Comment by dpvc** [comment]
I've also added code that prevents typeset math from attempting to be removed when the container is no longer in the document (e.g., the container had `innerHTML` set and the typeset math is now orphaned).

File: ts/output/common/OutputJax.ts [hunk 1]
```diff
@@ -296,9 +300,11 @@ export abstract class CommonOutputJax<
         //
         for (const math of html.math) {
             const node = adaptor.parent(math.start.node);
-            const map = domMaps[math.display? 1 : 0];
-            if (!map.has(node)) {
-                map.set(node, this.getTestElement(node, math.display));
+            if (node && math.state() < STATE.METRICS) {
+                const map = domMaps[math.display? 1 : 0];
```
Comment (zorkow) [inline ts/output/common/OutputJax.ts]: Missing space before ?

Comment (dpvc) [reply ts/output/common/OutputJax.ts]: Fixed.

**Pull request merged**